SPEC.md

```diff
@@ -2603,6 +2604,23 @@ Integer ylen = length(ys) # 3
 Integer zlen = length(zs) # 0
 ```
 
+## Array[X] flatten(Array[Array[X]])
+
+Given an array of arrays, the `flatten` function concatenates all the
+member arrays in the order to appearance to give the result. It does not
+deduplicate the elements. For example:
```

Comment:
Comment that arrays nested more deeply than 2 must be flattened twice (or more) to get down to an unnested `Array[X]`